crates/sqlbuiltins/src/functions/table/excel.rs

```diff
@@ -21,6 +23,22 @@ impl ConstBuiltinFunction for ExcelScan {
     const EXAMPLE: &'static str =
         "SELECT * FROM read_excel('file:///path/to/file.xlsx', sheet_name => 'Sheet1')";
     const FUNCTION_TYPE: FunctionType = FunctionType::TableReturning;
+    fn signature(&self) -> Option<Signature> {
+        let options: Fields = vec![
+            Field::new("sheet_name", DataType::Utf8, true),
+            Field::new("infer_rows", DataType::UInt64, true),
+            Field::new("has_header", DataType::Boolean, true),
+        ]
+        .into_iter()
+        .collect();
```

Comment:
From is implemented for `Vec<Field>` and Fields, so I think FromIterator handlers, which might make this simpler?

To be clear, this is fine, but I definitely stopped and read this three times because it *looks* wrong. "why are we making a vector, converting it to an iterator and collecting it back into a vector without doing anything else?" and while it compiles and does the thing, mds

If it's easy enough to change I think that would make it better and less likely to confuse other people in the future.